### PR TITLE
Modifying region acquisition from env variable

### PR DESCRIPTION
**Pull request by trwalke**
Fixes # .
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2803#

**Changes proposed in this request**
Modifying region acquisition from env variable

**Testing**

**Performance impact**

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs [hunk 1]
```diff
@@ -179,7 +179,6 @@ public async Task SuccessfulResponseFromRegionalizedAuthorityAsync()
         }
 
         [DataTestMethod]
-        [DataRow("Region with spaces")]
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs:182]: Regions with spaces are no longer invalid as we account for them in RegionManager.cs

Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs:182]: nit: not much point for a having a DataTestMethod if there is only 1 data row ...

File: src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs [hunk 2]
```diff
@@ -160,7 +160,7 @@ private async Task<RegionInfo> DiscoverAndCacheAsync(string azureRegionConfig, I
 
         private async Task<RegionInfo> DiscoverAsync(ICoreLogger logger, CancellationToken requestCancellationToken)
         {
-            string region = Environment.GetEnvironmentVariable("REGION_NAME");
+            string region = Environment.GetEnvironmentVariable("REGION_NAME")?.Replace(" ", string.Empty)?.ToLower();
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs:163]: Please add test that shows that this transformation is correct.

**Review by SameerK-MSFT — CHANGES_REQUESTED** [review]

File: src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs (hunk 2 above)
Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs:163]: @SameerK-MSFT, what happens if the environment variable does not exist?

File: src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs [hunk 3]
```diff
@@ -160,7 +160,7 @@ private async Task<RegionInfo> DiscoverAndCacheAsync(string azureRegionConfig, I
 
         private async Task<RegionInfo> DiscoverAsync(ICoreLogger logger, CancellationToken requestCancellationToken)
         {
-            string region = Environment.GetEnvironmentVariable("REGION_NAME");
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs:163]: `toLower` is culture sensitive, please use `toLowerInvariant` https://stackoverflow.com/questions/6225808/string-tolower-and-string-tolowerinvariant

File: src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs:163]: I just tested and it looks like @SameerK-MSFT  is right. According to the docs https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and- the null-conditional operator is short-circuiting, so the `ToLower()` never executes. 

Great question, I didn't know about this.

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs:163]: Thanks @SameerK-MSFT and @bgavrilMS : I did not know that was the way it worked!

**Review by SameerK-MSFT — APPROVED** [review]

**Review by pmaytak — APPROVED** [review]

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**